File: pi/rates.py
#!/usr/bin/env python3
"""Energy rate model.

Seattle City Light, Small General Service schedule (per May 2026 bill).
Flat $/kWh — no TOU peak/off-peak split, no seasonal variation.

If the service ever moves to a TOU schedule, reintroduce a get_rate(dt)
that branches on hour/season; callers already pass a datetime.
"""

from datetime import datetime

ENERGY_RATE = 0.1241        # $/kWh (Apr 2026+; pre-Apr was 0.1291)
BASE_CHARGE_DAILY = 0.83    # $/day fixed service charge (≈$47.31 / 57 days)


def get_rate(_dt: datetime) -> float:
    """Return $/kWh rate. Flat — datetime is accepted for API compatibility."""
    return ENERGY_RATE


def cost_for_kwh(kwh: float, dt: datetime) -> float:
    """Energy cost in dollars for the given kWh."""
    return kwh * get_rate(dt)


def is_peak(_dt: datetime) -> bool:
    """No TOU on this plan."""
    return False

File: web/lib/rates.ts
import { DAY_MS } from "./rollup";

/**
 * Flat energy rate model — Seattle City Light "Small General Energy" schedule
 * (per the May 2026 bill: $0.1241/kWh + $0.83/day base charge). No TOU/tiered
 * split modeled here; plan confirmation is still open (see CLAUDE.md Next
 * Steps), and `pi/rates.py` — the daily report's rate model — is the
 * source of truth to keep in sync with if the plan changes.
 */
export const ENERGY_RATE_PER_KWH = 0.1241; // $/kWh
export const BASE_CHARGE_PER_DAY = 0.83; // $/day fixed service charge

/** Energy cost in dollars for the given kWh at the flat rate. */
export function costForKwh(kwh: number): number {
  return kwh * ENERGY_RATE_PER_KWH;
}

/** Fixed service charge prorated over an arbitrary window length. */
export function proratedBaseCharge(windowMs: number): number {
  return (windowMs / DAY_MS) * BASE_CHARGE_PER_DAY;
}
